Samples/StyleTransfer/AppViewModel.cs

```diff
@@ -0,0 +1,546 @@
+using Windows.UI.Xaml.Media.Imaging;
+using Windows.UI.Xaml.Media;
+using Windows.Media.MediaProperties;
+using Windows.Storage.Pickers;
+using Windows.UI.Xaml.Controls;
+using System.IO;
+using Windows.System.Display;
+using Windows.UI.Core;
+using GalaSoft.MvvmLight.Threading;
+using System.Threading;
+using StyleTransferEffectCpp;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Devices.Enumeration;
+
+namespace StyleTransfer
+{
+    class AppViewModel : INotifyPropertyChanged
+    {
+        public AppViewModel()
+        {
+            _appModel = new AppModel();
+            SaveCommand = new RelayCommand(SaveOutput);
+            SetMediaSourceCommand = new RelayCommand<string>(async (x) => await SetMediaSource(x));
+            SetModelSourceCommand = new RelayCommand(async () => await SetModelSource());
+            ChangeLiveStreamCommand = new RelayCommand(async () => await ChangeLiveStream());
+
+            _inputSoftwareBitmapSource = new SoftwareBitmapSource();
+            _outputSoftwareBitmapSource = new SoftwareBitmapSource();
+            _saveEnabled = true;
+            NotifyUser(true);
+
+            m_notifier = new StyleTransferEffectCpp.StyleTransferEffectNotifier();
+            m_notifier.FrameRateUpdated += async (_, e) =>
+            {
+                await DispatcherHelper.RunAsync(() =>
+                {
+                    RenderFPS = e;
+                });
+
+            };
+            _useGpu = false;
+            isPreviewing = false;
+        }
+
+        // Media capture properties
+        public Windows.Media.Capture.MediaCapture _mediaCapture;
+        private bool isPreviewing;
+        private DisplayRequest displayRequest = new DisplayRequest();
+        DeviceInformationCollection devices;
+        System.Threading.Mutex Processing = new Mutex();
+        private float _renderFPS;
+        StyleTransferEffectCpp.StyleTransferEffectNotifier m_notifier;
+
+        // Style transfer effect properties
+        private LearningModel m_model = null;
+        private LearningModelDeviceKind m_inferenceDeviceSelected = LearningModelDeviceKind.Default;
+        private LearningModelSession m_session;
+        private LearningModelBinding m_binding;
+        private string m_inputImageDescription;
+        private string m_outputImageDescription;
+        // Activatable Class ID of the video effect. 
+        private String _videoEffectID = "StyleTransferEffectCpp.StyleTransferEffect";
+
+        // Image style transfer properties
+        uint m_inWidth, m_inHeight, m_outWidth, m_outHeight;
+        private string _DefaultImageFileName = "Input.jpg";
+
+        // User notifiaction properties
+        private bool _succeed;
+        private SolidColorBrush _successColor = new SolidColorBrush(Windows.UI.Colors.Green);
+        private SolidColorBrush _failColor = new SolidColorBrush(Windows.UI.Colors.Red);
+
+        private AppModel _appModel;
+        public AppModel CurrentApp

```

Comment:
>CurrentApp [](start = 24, length = 10)

Model

Reply:
Conflict with LearningModel. I can rename to have appModel and learningModel to better distinguish the two, though.